src/main/python/image_preprocessing/template_matching.py

```diff
@@ -29,26 +35,25 @@ def identify(src, template):
         """
 
         # load the source and template image
-        source = cv2.imread(src)
-        val = []
-        for (h, temp, t, n) in template:
-            templ = cv2.imread(temp)
-            r = h / source.shape[1]
-            dim = (h, int(source.shape[0] * r))
-            resized = cv2.resize(source, dim, interpolation=cv2.INTER_AREA)
+        source = src
+        template_objects = []
+        for (original_template_image_width, template_path, threshold, object_identifier) in template:
+            template_image = cv2.imread(template_path)
 
-            (tempH, tempW) = templ.shape[:2]
+            ratio = original_template_image_width / source.shape[1]
+            dimension = (original_template_image_width, int(source.shape[0] * ratio))
+            resized = cv2.resize(source, dimension, interpolation=cv2.INTER_AREA)
 
             # find the template in the source image
-            result = cv2.matchTemplate(resized, templ, cv2.TM_CCOEFF_NORMED)
+            result = cv2.matchTemplate(resized, template_image, cv2.TM_CCOEFF_NORMED)
 
-            (minVal, maxVal, minLoc, (x, y)) = cv2.minMaxLoc(result)
+            (_, maximum_value, minLoc, (x, y)) = cv2.minMaxLoc(result)
 
-            val = val + [(maxVal, t, n)]
+            template_objects = template_objects + [(maximum_value, threshold, object_identifier)]
 
-        for (m, s, n) in val:
-            if (m > s):
-                print(n)
-                return {'type': n}
+        for (max, threshold_value, object_type) in template_objects:
```

Comment:
At the moment none. The idea behind it was that I could sort the list to get the best match otherwise we could get a sub optimal value. As was stated in my todo of the function. I suppose we can change it until we see that the sorting will be necessary.